internal/jobs/launch_instance_aws.go

```diff
@@ -0,0 +1,91 @@
+package jobs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients/ec2"
+	"github.com/RHEnVision/provisioning-backend/internal/clients/sts"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/lzap/dejq"
+)
+
+type LaunchInstanceAWSTaskArgs struct {
+	AccountID     int64  `json:"account_id"`
+	PubkeyID      int64  `json:"pubkey_id"`
+	ReservationID int64  `json:"reservation_id"`
+	ImageID       string `json:"image_id"`
+	Amount        int32  `json:"amount"`
+	InstanceType  string `json:"instance_type"`
+	ARN           string `json:"arn"`
+	PubkeyHandle  string `json:"pubkey_handle"`
+}
+
+func EnqueueLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
+	logger := ctxval.Logger(ctx)
+	logger.Debug().Interface("arg", args).Msgf("Enqueuing launch instance AWS job: %+v", args)
+
+	pj := dejq.PendingJob{
+		Type: TypeLaunchInstanceAws,
+		Body: args,
+	}
+	err := Queue.Enqueue(ctx, pj)
+	if err != nil {
+		return fmt.Errorf("unable to enqueue: %w", err)
+	}
+
+	return nil
+}
+
+func HandleLaunchInstanceAWS(ctx context.Context, job dejq.Job) error {
+	ctxLogger := ctxval.Logger(ctx)
+	ctxLogger.Debug().Msg("Started launch instance AWS job")
+
+	args := LaunchInstanceAWSTaskArgs{}
+	err := job.Decode(&args)
+	if err != nil {
+		ctxLogger.Error().Err(err).Msg("unable to decode arguments")
+		return fmt.Errorf("unable to decode args: %w", err)
+	}
+
+	ctx = ctxval.WithAccountId(ctx, args.AccountID)
+	logger := ctxLogger.With().Int64("reservation", args.ReservationID).Logger()
+	logger.Info().Interface("args", args).Msg("Processing launch instance AWS job")
+
+	// upload to cloud with a tag
+	client := ec2.NewEC2Client(ctx)
+	stsClient, err := sts.NewSTSClient(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot initialize sts client: %w", err)
+	}
+
+	crd, err := stsClient.AssumeRole(args.ARN)
+	if err != nil {
+		return fmt.Errorf("cannot assume role: %w", err)
+	}
+
+	newEC2Client, err := client.CreateEC2ClientFromConfig(crd)
+	if err != nil {
+		return fmt.Errorf("cannot create new ec2 client from config: %w", err)
+	} //get the info needed to launch and instance and pass it
+	//TODO: NEEDS THE RESOURCE NAME LIKE "lzap-rsa-2010-{{$randomInt}}"
+	// Should be added to args.PubkeyHandle, the name is not saved?
+	_, err = newEC2Client.RunInstances(ctx, args.Amount, types.InstanceType(args.InstanceType), args.ImageID, "lzap-rsa-2010-{{$randomInt}}")
```

Comment:
the `imageID` from the reservation request is an inner identifier of the image in image-builder service,
we should [fetch](https://github.com/RHEnVision/provisioning-backend/pull/116/files#diff-84c4846afa93893d6a3a76ddb49781b320530975ebb9c6e781a320f09ca97cd6R30) the ami from image-builder via its client

Reply:
the image Id written here is the AWS image identifier like in the example: 
"ami-0cff7528ff583bf9a"

Reply:
Should we name it AMI then? and make use of Amir's pr in the service I guess (simiar to what we do with ARN)

Reply:
I believe we should send image_id to the API, but I'd resolve that into AMI before queing the job, but happy to be convinced that we should push these to the job, but we are missing the auth header in the context, so we just resolved `source_id -> ARN` in the service and I guess this is the same case.
But I'd agree that it's kinda being lazy and pushing too much to the service can make it too slow.

Reply:
Not sure what did you mean.
I don't use image_id from image builder, I just use the AMI. 
Don't mind changing it to AMI instead of image_id if we all agree.

Reply:
I agree that all jobs should only need to access our database and the cloud, therefore resolving the imagebuilder ID to AMI would be the best to do before the job is enqueued.

Reply:
yeah, agreed, so I'd name it `ami` to be clear